src/index.ts

```diff
@@ -79,26 +80,32 @@ export type SagaPromiseActionCreator<V, P, T extends string, TA extends PayloadA
 
 export type SagaPromisePreparedActionCreator<V, T extends string, TA extends PrepareAction<any>> = SagaPromiseActionCreator<V, ReturnType<TA>["payload"], T, _ActionCreatorWithPreparedPayload<TA, T>>;
 
+let promiseResolutionMap: PromiseResolutionMap<any> | undefined;
```

Comment:
This is an absolute no-no. We cannot shift the `promiseResolution` to the scope of the module nor we ever should let multiple actions reassign a module-scoped variable, here namely `promiseResolution`.

Reply:
`promiseResolution`is absolutely tight to one promsie action, imagine it as 1:1 relationship.

Reply:
@teneko what do you suggest?

Reply:
This sounds like the initially created promise action has not surpassed the middleware? :o
Could you create an absolute minimal reproduction of your actual problem?

Reply:
We come to a point I cannot tell for sure where the actual problem is lying. Does it help you to explain the problem, by inspecting the logic implemented here: https://github.com/teneko/redux-saga-promise/blob/d148fb427b3de17f1dcb9edaf6f56214bba4489d/src/index.ts#L152 ?

Reply:
@teneko have you had the time to look at my codepen yet?

Reply:
I am going to inspect this today ! Sorry for the delay.

Reply:
No worries

Reply:
I need to shift this to saturday. Lot of work.

Reply:
Thanks for your input and work so far ! This needs definitvely to be investigated and solved !

Reply:
Sure no worries

Reply:
Oh damn, I think you misunderstood how actions work. :(

Similiar to the thing you just assumed reassigning to `let promsieResolutionMap`is fine you unfortunately thought that you can also use a **new** action, by using the action creator in a saga handler, this is not how actions are handled in a saga handler.

As soon as you dispatch a **new** action, by doing `promiseAction()` then this action, and by this I mean the instance you just created is pass-throughed the redux saga middleware and therefore redux sagas and therefore also passed in your defined handler method.

So please look at the following code:

```typescript
import createSagaMiddleware from "redux-saga";
import { call, takeLatest } from "redux-saga/effects";
import { configureStore, createReducer } from "@reduxjs/toolkit";
import {
  promiseMiddleware,
  promiseActionFactory,
  resolvePromiseAction,
} from "./index";

// Create promise action **creator**
const promiseAction = promiseActionFactory<string>().create("promiseAction");

function* handleTriggerAction(action: any) {
  try {
    yield call(resolvePromiseAction, action, "works");
    // or
    // yield call(promiseAction.sagas.resolve, action, "works");
  } catch (e) {
    yield call(promiseAction.sagas.reject, action, e.message);
  }
}

function handleResolvedAction(action: typeof promiseAction.types.resolvedAction) {
  console.log(`${action.payload} (hello from saga)`);
}

function* watcher() {
  yield takeLatest(promiseAction, handleTriggerAction);
  yield takeLatest(promiseAction.resolved, handleResolvedAction);
}

const reducer = createReducer<string>("<store-yet-uninitialized>", (builder) => {
  builder.addCase(promiseAction.trigger, (state) => state);
  builder.addCase(promiseAction.resolved, (_unused_state, action) => `${action.payload} (hello from store)`);
  builder.addCase(promiseAction.rejected, (state) => state);
});

const sagaMiddleware = createSagaMiddleware();

// createStore is deprecated, so use:
const store = configureStore({
  reducer,
  middleware: [promiseMiddleware, sagaMiddleware],
});
// or
// const store = legacy_createStore(reducer, middleware);

sagaMiddleware.run(watcher);

store.subscribe(() => {
  console.log(store.getState());
});

// promiseAction() leads to a NEW instance,
// which first gets transformed to a fully qualified promise action by OUR middleware
// and then THIS NEW INSTANCE is passed through the redux saga :)
store.dispatch(promiseAction()).then((result) => console.log(`${result} (hello from promise)`));
```

You simply can paste this code to "/src/testCase.ts"

and run it by:

```
$ npm install -g ts-node
$ ts-node /src/testCase.ts
```

You will see this output:

```txt
<store-yet-uninitialized>
works (hello from store)
works (hello from saga)
works (hello from promise)
```